src/HookTarget/BaseHookTarget.sol

```diff
@@ -0,0 +1,42 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+pragma solidity >=0.8.0;
+
+import {GenericFactory} from "evk/GenericFactory/GenericFactory.sol";
+import {IHookTarget} from "evk/interfaces/IHookTarget.sol";
+
+/// @title BaseHookTarget
+/// @custom:security-contact security@euler.xyz
+/// @author Euler Labs (https://www.eulerlabs.com/)
+/// @notice Base contract for hook targets associated with the vaults deployed by the EVK factory.
+abstract contract BaseHookTarget is IHookTarget {
+    /// @notice The EVault factory contract.
+    GenericFactory public immutable eVaultFactory;
+
+    /// @notice Constructor
+    /// @param _eVaultFactory The address of the EVault factory contract which deployed the vaults associated with
+    /// this hook target.
+    constructor(address _eVaultFactory) {
+        eVaultFactory = GenericFactory(_eVaultFactory);
+    }
+
+    /// @inheritdoc IHookTarget
+    /// @dev This function returns the expected magic value only if the caller is a proxy deployed by the recognized
+    /// EVault factory.
+    function isHookTarget() external view override returns (bytes4) {
+        if (eVaultFactory.isProxy(msg.sender)) return this.isHookTarget.selector;
+        else return 0;
```

Comment:
Maybe just revert here. It more closely follows what a non-hook contract would do, and in theory the isHookTarget selector could be `bytes4(0)` :)

Reply:
The reason for not reverting here is that we expect the vault to revert with a standard `E_NotHookTarget` error if unexpected selector is returned. Reverting here would mean that one can no longer rely on this standard error.

Reply:
What relies on it, the perspective verification?

It's actually interesting, because a hook could revert with whatever it wanted (including that error or others). If it's propagated upwards, I guess perspective could not tell the difference...

Reply:
We don't rely on it, but somebody else can potentially? Hard to say what would be better here, I guess it doesn't matter that much.